compiler/rustc_resolve/src/late.rs

```diff
@@ -2401,32 +2402,72 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
                 });
             }
 
-            ItemKind::Static(box ast::StaticItem { ref ty, ref expr, .. })
-            | ItemKind::Const(box ast::ConstItem { ref ty, ref expr, .. }) => {
+            ItemKind::Static(box ast::StaticItem { ref ty, ref expr, .. }) => {
                 self.with_static_rib(|this| {
                     this.with_lifetime_rib(LifetimeRibKind::Elided(LifetimeRes::Static), |this| {
                         this.visit_ty(ty);
                     });
                     this.with_lifetime_rib(LifetimeRibKind::Elided(LifetimeRes::Infer), |this| {
                         if let Some(expr) = expr {
-                            let constant_item_kind = match item.kind {
-                                ItemKind::Const(..) => ConstantItemKind::Const,
-                                ItemKind::Static(..) => ConstantItemKind::Static,
-                                _ => unreachable!(),
-                            };
                             // We already forbid generic params because of the above item rib,
                             // so it doesn't matter whether this is a trivial constant.
                             this.with_constant_rib(
                                 IsRepeatExpr::No,
                                 ConstantHasGenerics::Yes,
-                                Some((item.ident, constant_item_kind)),
+                                Some((item.ident, ConstantItemKind::Static)),
                                 |this| this.visit_expr(expr),
                             );
                         }
                     });
                 });
             }
 
+            ItemKind::Const(box ast::ConstItem { ref generics, ref ty, ref expr, .. }) => {
+                // FIXME(fmease): #prMergeBlocker What should the ordering of the ribs be?
+                // Right now, it's STATIC <-- LIFETIME#0 <-- GENERIC <-- LIFETIME#1 <-- {,CONSTANT}
```

Comment:
rephase that, make it less awkward to parse. and use more meaningful names for `LT#N`